src/sinks/mqtt/config.rs

```diff
@@ -0,0 +1,176 @@
+use std::time::Duration;
+
+use codecs::JsonSerializerConfig;
+use rand::{thread_rng, Rng};
+use rumqttc::{MqttOptions, TlsConfiguration, Transport};
+use snafu::ResultExt;
+use vector_config::configurable_component;
+
+use crate::{
+    codecs::EncodingConfig,
+    config::{AcknowledgementsConfig, GenerateConfig, Input, SinkConfig, SinkContext},
+    sinks::{
+        mqtt::sink::{MqttConnector, MqttError, MqttSink, TlsSnafu},
+        Healthcheck, VectorSink,
+    },
+    tls::{MaybeTlsSettings, TlsEnableableConfig},
+};
+
+/// Configuration for the `mqtt` sink
+#[configurable_component(sink("mqtt"))]
+#[derive(Clone, Debug)]
+pub struct MqttSinkConfig {
+    /// MQTT server address
+    pub host: String,
+
+    /// TCP port
+    #[serde(default = "default_port")]
+    pub port: u16,
+
+    /// MQTT username
+    pub user: Option<String>,
+
+    /// MQTT password
+    pub password: Option<String>,
+
+    /// MQTT client ID
+    #[serde(default = "default_client_id")]
+    pub client_id: String,
+
+    /// Connection keep-alive interval
+    #[serde(default = "default_keep_alive")]
+    pub keep_alive: u16,
+
+    /// Clean MQTT session on login or not
+    #[serde(default = "default_clean_session")]
+    pub clean_session: bool,
+
+    #[configurable(derived)]
+    pub tls: Option<TlsEnableableConfig>,
+
+    /// MQTT publish topic (templates allowed)
+    pub topic: String,
```

Comment:
```suggestion
    pub topic: Template,
```

We can set it as a `Template` here.

Reply:
Fixed.